appservice/src/createAppService/AppServicePlanSkuStep.ts

```diff
@@ -16,31 +16,47 @@ type ExtendedSkuDescription = WebSiteManagementModels.SkuDescription & { label?:
-        let skus: ExtendedSkuDescription[] = wizardContext.advancedCreation ? this.getRecommendedSkus().concat(this.getAdvancedSkus()) : this.getRecommendedSkus();
-        if (wizardContext.newSiteKind === AppKind.functionapp) {
-            skus.push(...this.getElasticPremiumSkus());
-        } else if (wizardContext.newSiteKind?.includes(AppKind.workflowapp)) {
-            skus = this.getWorkflowStandardSkus();
-        }
-
+        let recommendedSkus = this.getRecommendedSkus();
         const regExp: RegExp | undefined = wizardContext.planSkuFamilyFilter;
         if (regExp) {
-            skus = skus.filter(s => !s.family || regExp.test(s.family));
+            recommendedSkus = recommendedSkus.filter(s => !s.family || regExp.test(s.family));
         }
-
-        const pricingTiers: IAzureQuickPickItem<WebSiteManagementModels.SkuDescription | undefined>[] = skus.map(s => {
+        const pricingTiers: IAzureQuickPickItem<WebSiteManagementModels.SkuDescription | undefined>[] = recommendedSkus.map(s => {
             return {
                 label: s.label || nonNullProp(s, 'name'),
                 description: s.description || s.tier,
-                data: s
+                data: s,
+                group: localize('recommendedSkusLabel', 'Recommended')
             };
         });
 
+        let advancedSkus: ExtendedSkuDescription[] = [];
+        if (wizardContext.advancedCreation) {
+            advancedSkus.push(...this.getAdvancedSkus());
+            if (regExp) {
+                advancedSkus = advancedSkus.filter(s => !s.family || regExp.test(s.family));
+            }
+            if (wizardContext.newSiteKind === AppKind.functionapp) {
+                advancedSkus.push(...this.getElasticPremiumSkus());
+            } else if (wizardContext.newSiteKind?.includes(AppKind.workflowapp)) {
+                advancedSkus = this.getWorkflowStandardSkus();
+            }
+
+            pricingTiers.push(...advancedSkus.map(s => {
+                return {
+                    label: s.label || nonNullProp(s, 'name'),
+                    description: s.description || s.tier,
+                    data: s,
+                    group: localize('additionalOptionsLabel', 'Additional Options'),
+                };
+            }));
+        }
+
         pricingTiers.push({ label: localize('ShowPricingCalculator', '$(link-external) Show pricing information...'), data: undefined, suppressPersistence: true });
 
         while (!wizardContext.newPlanSku) {
             const placeHolder = localize('pricingTierPlaceholder', 'Select a pricing tier');
-            wizardContext.newPlanSku = (await wizardContext.ui.showQuickPick(pricingTiers, { placeHolder, suppressPersistence: true })).data;
+            wizardContext.newPlanSku = <WebSiteManagementModels.SkuDescription | undefined>(await wizardContext.ui.showQuickPick(pricingTiers, { placeHolder, suppressPersistence: true, enableGrouping: wizardContext.advancedCreation })).data;
```

Comment:
This cast should not be necessary. The type of `data` should already be `WebSiteManagementModels.SkuDescription | undefined` because the type of `pricingTiers` is `IAzureQuickPickItem<WebSiteManagementModels.SkuDescription | undefined>[]`

Reply:
This is to fix the lint error that was occurring before. Either this or getting a "Unsafe assignment of an any value" lint error.

Reply:
If I take your branch and remove the cast:
https://github.com/microsoft/vscode-azuretools/commit/5e5ca3559bdb39caecf8537902e992dfb4038f0b

The build passes with no lint error:
https://dev.azure.com/ms-azuretools/AzCode/_build/results?buildId=34421&view=logs&j=0da5d1d9-276d-5173-c4c4-9d4d4ed14fdb&t=4c091adb-ab1c-50df-3a5f-280100199621